doozerlib/distgit.py

```diff
@@ -331,146 +331,110 @@ def image_build_method(self):
+            if not self.runtime.group_config.doozer_feature_gates.odcs_aggressive:
+                # Doozer's odcs_aggressive feature gate is off, disable ODCS mode for this image
+                return config_overrides
+            self.logger.warning("Enforce ODCS auto mode because odcs_aggressive feature gate is on")
 
-        package_mode = odcs.packages.get('mode', 'auto').lower()
+        package_mode = odcs.packages.get('mode', 'auto')
         valid_package_modes = ['auto', 'manual']
         if package_mode not in valid_package_modes:
-            raise ValueError('odcs.packages.mode must be one of {}'.format(str(valid_package_modes)))
+            raise ValueError('odcs.packages.mode must be one of {}'.format(', '.join(valid_package_modes)))
 
-        config = source_container_yaml
-        if 'compose' not in config:
-            config['compose'] = {}
-
-        if config['compose'].get('packages', []):
-            package_mode = 'pre'  # container.yaml with packages was given
-        compose_content = {
-            'packages': [],
-            'pulp_repos': True
-        }
+        # generate container.yaml content for ODCS
+        config = {}
+        if self.has_source():  # if upstream source provides container.yaml, load it.
+            source_container_yaml = os.path.join(self.source_path(), 'container.yaml')
+            if os.path.isfile(source_container_yaml):
+                with open(source_container_yaml, 'r') as scy:
+                    config = yaml.full_load(scy)
 
         # ensure defaults
-        for k, v in compose_content.iteritems():
-            if k not in config['compose']:
-                config['compose'][k] = v
-
-        # always overwrite platforms so we control it
-        config['platforms'] = {'only': arches}
-
-        config = Model(config)
+        config.setdefault('compose', {}).setdefault('pulp_repos', True)
 
+        # create package list for ODCS, see https://osbs.readthedocs.io/en/latest/users.html#compose
         if package_mode == 'auto':
-            packages = []
-            for rpm in self.metadata.get_rpm_install_list():
-                res = []
-                # ODCS is fine with a mix of packages that are only available in a single arch
-                for arch in self.metadata.runtime.arches:
-                    res_arch = check_rpm_buildroot(rpm, self.branch, arch)
-                    if res_arch:
-                        if isinstance(res_arch, list):
-                            res.extend(res_arch)
-                        else:
-                            res.append(res_arch)
-
-                res = list(set(res))
-                if res:
-                    packages.extend(res)
-
-            if odcs.packages.list:
-                config.compose.packages.extend(odcs.packages.list)
-
-            if odcs.packages.exclude:
-                exclude = set(odcs.packages.exclude)
+            if config["compose"].get("packages") is list:
```

Comment:
shouldn't this be detecting if it is an instance of `list`, not if it `is list`?
```suggestion
            if isinstance(config["compose"].get("packages"), list):
```
unit tests would help :)